### DISCUSSION TITLE
Ability to control song clips from keyboard mode

### DISCUSSION THREAD

**Discussion by michaelforrest** [SynthstromAudible/DelugeFirmware#2229] [Ideas]
I'd like to be in a clip and when I toggle keyboard mode, I see the song mode loops on the right hand side so I can play a synth and toggle other parts from the same place. 

Only UI problem I can see is scrolling the song columns separately from the keyboard, but maybe it could be the vertical push-in and scroll. 

I'm happy to have a crack at the code if anybody could tell me where to look?

**Comment by seangoodvibes** [comment]
Hey @michaelforrest so someone's actually looking at this already!

You can follow their progress here:

https://github.com/SynthstromAudible/DelugeFirmware/pull/1545

**Reply by michaelforrest** [reply]
Thanks Sean, I'll take a look!
